### PR TITLE
Get SCM Predictions and LOIO Probabilities

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, predictions and leave-one-image-out probabilities are derived for the single-class models.

For predictions, we simply load in the model for a particular phenotypic class, model type (final or shuffled baseline), and feature type (CP, DP, CP_and_DP) then derive model predictions for cells from both data subsets (train and test). Predictions are saved to `predictions/`.

The LOIO evaluation procedure is as follows:
- Load in entire MitoCheck labeled cell dataset (from [labeled_data.csv.gz](0.download_data/data/labeled_data.csv.gz)).
- For each image in MitoCheck labeled cell dataset (as specified by the `Metadata_DNA` field):
    - Train a logistic regression model with optimal hyperparameters (`C` and `l1_ratio`, saved with model data in [models/](2.train_model/models/)) on every cell that is **not** in the specific image.
    - Predict probabilities on every cell that **is** in the specific image.

The probabilities are saved to `LOIO_probas/`.

### CODE REVIEW

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
+from joblib import load
+import itertools
 
 import pandas as pd
 
 from sklearn.linear_model import LogisticRegression
 from sklearn.utils import parallel_backend
 from sklearn.exceptions import ConvergenceWarning
-from joblib import load
+from sklearn.metrics import f1_score
```
Comment (axiomcura) [inline 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py]:
```suggestion
import pathlib
import warnings
import itertools

from joblib import load
import pandas as pd

from sklearn.linear_model import LogisticRegression
from sklearn.utils import parallel_backend
from sklearn.exceptions import ConvergenceWarning
from sklearn.metrics import f1_score
```
According to the [documentation](https://peps.python.org/pep-0008/#imports), standard library imports should be first then related third party imports. I do not think that `joblib` is part of the standard library. 

Feel free to remove this suggestion if this incorrect.

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
 
 import sys
```
Comment (axiomcura) [inline 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py]: Consider moving this to the top of all your standard library imports. Refer to the second bullet point of the [documentation](https://peps.python.org/pep-0008/#imports).

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py [hunk 3]
```diff
@@ -41,21 +50,22 @@
...
 
+# ### Get LOIO probabilities
+# 
+
 # In[4]:
 
 
 # directory to load the models from
-models_dir = pathlib.Path("../2.train_model/models/")
+models_dir = pathlib.Path("../2.train_model/models/multi_class_models")
```
Comment (axiomcura) [inline 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py:59]:
```suggestion
models_dir = pathlib.Path("../2.train_model/models/multi_class_models").reslove(strict=True)
```
This may be out of scope for this PR. Recently the `pathlib.Path.resolve()` has been really usefull in handling different paths (i.e symlink paths). The `strict` parameters checks if the provided path exists before conducting any down stream functions. 

Here's the [documentation](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve), may be useful for future implementations for path handling!

**Review by axiomcura — APPROVED** [review]
Good PR! Level some comments and suggestions. 

Feel free to ask me anything!

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py (hunk 1 above)
Comment (roshankern) [reply 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py]: Ah, this is correct. Great catch!

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py (hunk 3 above)
Comment (roshankern) [reply 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py:59]: I think it is probably outside the scope of this PR, but thank you for letting me know about this function.

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py (hunk 2 above)
Comment (ghost) [reply 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py]: Done in https://github.com/WayScience/phenotypic_profiling_model/pull/29/commits/ba35da85f8ae291547b956b96edde5312a76fe19

File: 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py (hunk 1 above)
Comment (ghost) [reply 3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py]: Added to jupyter notebook in https://github.com/WayScience/phenotypic_profiling_model/pull/29/commits/ba35da85f8ae291547b956b96edde5312a76fe19
